### PR TITLE
Fix GDNative C++ SConstruct for OSX

### PR DESCRIPTION
**Pull request by vonagam**
Without `-std=c++17` compiling will fail with errors shown [here](https://godotengine.org/qa/65616/problems-compiling-gdnative-c-example-on-osx).

### CODE REVIEW

File: tutorials/plugins/gdnative/files/cpp_example/SConstruct [hunk 1]
```diff
@@ -50,10 +50,10 @@ if env['platform'] == "osx":
     env['target_path'] += 'osx/'
     cpp_library += '.osx'
     if env['target'] in ('debug', 'd'):
-        env.Append(CCFLAGS=['-g', '-O2', '-arch', 'x86_64'])
+        env.Append(CCFLAGS=['-g', '-O2', '-arch', 'x86_64', '-std=c++17'])
```
Comment (akien-mga) [inline tutorials/plugins/gdnative/files/cpp_example/SConstruct:53]: `CCFLAGS` are the flags for both the C and C++ compilers, so we can't pass `-std=c++17` to the C compiler (Clang would normally raise an error, but I guess Apple Clang is more lenient if it works for you).

This should be:
```
env.Append(CXXFLAGS=['-std=c++17'])
```
to only affect C++.

Comment (vonagam) [reply tutorials/plugins/gdnative/files/cpp_example/SConstruct]: How about now? Moved common flags from conditionals for osx and linux (like it is for windows).

**Comment by vonagam** [comment]
Anything else holding this back?

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**